api/python/quilt3/packages.py

```diff
@@ -910,11 +911,19 @@ def set_dir(self, lkey, path=None, meta=None, update_policy="incoming"):
             src_path = src.path
             if src.basename() != '':
                 src_path += '/'
-            objects, _ = list_object_versions(src.bucket, src_path)
+            try:
+                objects, _ = list_object_versions(src.bucket, src_path)
+            except botocore.exceptions.ClientError as e:
```

Comment:
it's still too broad
```suggestion
            except botocore.exceptions.ClientError as e:
                if e.response["Error"]["Code"] != "AccessDenied":
                    raise
```

Reply:
Fixed in new PR